src/main/java/com/thirdcc/webapp/web/rest/EventResource.java

```diff
@@ -130,4 +130,19 @@ public EventResource(EventService eventService) {
         eventService.delete(id);
         return ResponseEntity.noContent().headers(HeaderUtil.createEntityDeletionAlert(applicationName, true, ENTITY_NAME, id.toString())).build();
     }
+
+    /**
+     {@code POST  /event/{eventId}/deactivate} : cancel "id" event.
+     *
+     * @param eventId the id of the eventDTO to cancel.
+     * @return the {@link ResponseEntity} with status {@code 200 (OK)} and 400 ()}.
+     */
+    @PutMapping("/event/{eventId}/deactivate")
+    public ResponseEntity<EventDTO> cancelEvent(@PathVariable Long eventId ){
```

Comment:
need add @preAuthorize annotation to only let current CC head to cancel the event

Reply:
@junzai97  if fixed please resolved conversation